### DISCUSSION TITLE
Is it possible to setup AIO on LAN behind FRP?

### DISCUSSION THREAD

**Discussion by zhou-wb** [nextcloud/all-in-one#611] [Questions] [answered]
- I want to setup nextcloud aio on my old PC running Ubuntu. However, there is no public IP address for my home router. So I rent a VPS which has a public IP and I use FRP[(Fast Reverse Proxy)](https://github.com/fatedier/frp) to expose my Ubuntu local server to the Internet and I also have a domain pointing to the VPS.
 - I wonder if it is possible to install the aio under this situation? What type of protocols/ports do I need to forward from the VPS to my home PC? I have read the [reverse-proxy.md](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md), but this seems to be a different situation? Thanks for any helps!

**Comment by szaimen** [comment]
Hi, in general it should be possible to use AIO in such a setup but I will not investigate this. Feel free to investigate yourself and post your findings here! Thanks!

**Reply by zhou-wb** [reply]
Thanks for your reply! I will test it when my domain is ready to use.

**Comment by szaimen** [comment, marked as answer]
You may be able to make it work by following the reverse proxy documentation and disabling the domain validation which is possible since v1.5.0.

**Reply by BoboYaung** [reply]
when i use nextcloud behind frp and nginx proxy manager, it show Bad Gateway when i use domain

**Comment by xZinio** [comment]
Hi, can you tell me if you made it work and if yes, how?
I'm really stuck at getting to Nextcloud in my browser through FRP, even though I already got an SSL certificate and got it configured my FRP config.

**Reply by BoboYaung** [reply]
I have made it work. You need to set local port 11000,and remote port as what you want.
